docs/02-app/01-building-your-application/02-data-fetching/03-server-actions-and-mutations.mdx

```diff
@@ -683,37 +685,77 @@ Remember to consider the [behavior and caveats](https://react.dev/reference/reac
 
 ### Error Handling
 
-When an error is thrown, it'll be caught by the nearest [`error.js`](/docs/app/building-your-application/routing/error-handling) or `<Suspense>` boundary on the client. We recommend using `try/catch` to return errors to be handled by your UI.
+#### User Errors
+
+Uncaught errors in Server Actions are considered to be internal server errors. As a result, in production, the error message is omitted and instead [a generic error with a digest](/docs/app/building-your-application/routing/error-handling#securing-sensitive-error-information) is sent to avoid leaking potentially sensitive information.
 
-For example, your Server Action might handle errors from creating a new item by returning a message:
+Therefore, if you want to return a user-friendly error message to be displayed in the UI, you should return a serializable object with the error message, rather than throwing an error:
 
 ```ts filename="app/actions.ts" switcher
 'use server'
 
-export async function createTodo(prevState: any, formData: FormData) {
-  try {
-    // Mutate data
-  } catch (e) {
-    throw new Error('Failed to create task')
+export async function signIn(formData: FormData) {
+  const passwordIsCorrect = await checkPassword(formData.get('password'))
+  if (!passwordIsCorrect) {
+    return { error: 'Incorrect password' }
   }
+  // ...
 }
 ```
 
 ```js filename="app/actions.js" switcher
 'use server'
 
-export async function createTodo(prevState, formData) {
-  try {
-    //  Mutate data
-  } catch (e) {
-    throw new Error('Failed to create task')
+export async function signIn(formData) {
+  const passwordIsCorrect = await checkPassword(formData.get('password'))
+  if (!passwordIsCorrect) {
+    return { error: 'Incorrect password' }
   }
+  // ...
 }
 ```
 
-> **Good to know:**
->
-> - Aside from throwing the error, you can also return an object to be handled by `useActionState`. See [Server-side validation and error handling](#server-side-form-validation).
+In the client, you can then use the returned object to display the error message. We recommend using the [`useActionState`](https://react.dev/reference/react/useActionState) hook to simplify this process. See [Server-side form validation](#server-side-form-validation) for an example.
+
+#### Server Errors
+
+When an **unexpected** error occurs, it'll be caught by the nearest [`error.js`](/docs/app/building-your-application/routing/error-handling) or `<Suspense>` boundary on the client. Therefore you can use `error.js` to let the user know that something has gone wrong.
+
+```tsx filename="app/dashboard/error.tsx" switcher
```

Comment:
Before the `error.js` example, we can add a minimum example of showing the try/catch in the server action.